### PR TITLE
Merge otlp collector examples

### PR DESCRIPTION
**Pull request by wtong98**
For issue #795 

This PR merges the OTLP example with the otel-collector example, adding a Prometheus backend to handle metrics.

### CODE REVIEW

File: example/otel-collector/README.md [hunk 1]
```diff
@@ -126,55 +158,53 @@ spec:
...
+```
+indicating that the Jaeger UI is available at
+[http://localhost:80](http://localhost:80). Navigate there in your favorite
+web-browser to view the generated traces.
+
+## Prometheus
+Unfortunately, the Prometheus operator doesn't provide a convenient
+out-of-the-box ingress route for us to use, so we'll use port-forwarding
+instead. Note: this is a quick-and-dirty solution for the sake of example.
+You *will* be attacked by shady people if you do this in production
```
Comment (MrAlias) [inline example/otel-collector/README.md:203]:
```suggestion
You *will* be attacked by shady people if you do this in production!
```

File: example/otel-collector/main.go [hunk 2]
```diff
@@ -23,53 +25,101 @@ import (
...
+	defer pusher.Stop() // pushes any last exports to the receiver
+
+	tracer := global.Tracer("test-tracer")
+	meter := global.Meter("test-meter")
+
+	// labels represent additional key-value descriptors that can be bound to a
+	// metric observer or recorder.
+	commonLabels := []kv.KeyValue{
+		kv.String("labelA", "chocolate"),
+		kv.String("labelB", "rasberry"),
```
Comment (MrAlias) [inline example/otel-collector/main.go:94]:
```suggestion
		kv.String("labelB", "raspberry"),
```

**Comment by MrAlias** [comment]
@wtong98 can you merge master into this and fix the impending conflict? I think this is ready to merge once that is done.

**Comment by wtong98** [comment]
@MrAlias All set! Thanks for taking a look!

**Pull request merged**